src/bundle.rs

```diff
@@ -354,6 +358,25 @@ impl<T: Authorization, V> Bundle<T, V> {
             .map(|a| a.to_instance(self.flags, self.anchor))
             .collect()
     }
+
+    /// Perform trial decryption of each action in the bundle with each of the
+    /// specified incoming viewing keys, and return the decrypted note contents
+    /// along with the index of the action from which it was derived.
+    pub fn decrypt_outputs_for_keys<'a>(
+        &self,
+        keys: &[IncomingViewingKey],
+    ) -> Vec<(usize, IncomingViewingKey, Note, Address, [u8; 512])> {
+        self.actions
+            .iter()
+            .enumerate()
+            .filter_map(|(idx, action)| {
+                let domain = OrchardDomain::for_action(action);
+                keys.iter().find_map(move |ivk| {
+                    try_note_decryption(&domain, ivk, action).map(|(n, a, m)| (idx, ivk.clone(), n, a, m))
```

Comment:
I initially thought this would lose information about which ivk was used to decrypt, but actually [try_note_decryption](https://github.com/zcash/librustzcash/blob/c7c79d266efe626eaa67e0997b3a3ca97dd2552e/components/zcash_note_encryption/src/lib.rs#L323-L335) includes the `Recipient` in the returned tuple, which I guess is sufficient.